src/integ_test_resources/flutter/amplify/github-cloudwatch-cdk/lib/createCanaryTestFailureAlarm.ts

```diff
@@ -0,0 +1,72 @@
+import { Duration } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { Alarm, ComparisonOperator, Metric, TreatMissingData, DimensionsMap, Stats } from 'aws-cdk-lib/aws-cloudwatch';
+
+const metricNamespace = "GithubCanaryApps";
+
+// Create the metric and alarms for tracking Github Canary failures.
+// These metrics are sent in Github action: amplify-flutter/amplify_canaries
+export function createCanaryTestFailureAlarm(scope: Construct) {
+
+  const buildFailureMetric = createMetric(
+    "BuildCanaryTestFailure",
+    {
+      "channel": "stable",
+      "label": "CanaryBuildFailures",
+    }
+  );
+
+  const E2EAndroidMetric = createMetric(
+    "E2ECanaryTestFailure",
+    {
+      "channel": "stable",
+      "platform": "android",
+      "label": "E2ECanaryFailures",
+    }
+  );
+
+  const E2EIosMetric = createMetric(
+    "E2ECanaryTestFailure",
+    {
+      "channel": "stable",
+      "platform": "ios",
+      "label": "E2ECanaryFailures",
+    }
+  );
+
+  createAlarm(scope, `build-canary-test-failure-alarm`, "Build Canary Test Failure Alarm", "Alarm triggered when the Github Action Canaries build step fails", buildFailureMetric);
+
+  createAlarm(scope, `e2e-android-canary-test-failure-alarm`, "E2E Android Canary Test Failure Alarm", "Alarm triggered when the Github Action Canaries E2E android step fails", E2EAndroidMetric);
+
+  createAlarm(scope, `e2e-ios-canary-test-failure-alarm`, "E2E iOS Canary Test Failure Alarm", "Alarm triggered when the Github Action Canaries E2E ios step fails", E2EIosMetric);
+
+  // TODO: Create Github issues as alarm actions
+
+  // TODO: Consider moving to brazil to allow SIM Ticket creation.
+  // reason: package 'SIMTicketAlarmAction' is an AWS internal package.
+
+}
+
+function createMetric(metricName: string, dimensions: DimensionsMap): Metric {
+  return new Metric({
+    metricName,
+    namespace: metricNamespace,
+    statistic: Stats.MAXIMUM,
+    period: Duration.hours(24),
```

Comment:
Good point.  Better to do 1 hour or even shorter since alarm status is only changed when a new metric is received and a new duration segment is entered.

Reply:
In this case we might as well do 5 minutes to ensure faster updates right?  Switching to that.

Reply:
sorry, uber nit here but we should do at least 15 min (or ideally an hour) bc in our case, I think a single run will submit 2 data points for build/stable (one for flutter version "any" and the other for 3.10.x [lowest]). One run could finish 5 min after the other, in which case, the second one succeeding would clear out error from failure in 1st one.

Reply:
good call, updated to 1 hour.